portfolio/src/main/webapp/script.js

```diff
@@ -297,3 +311,72 @@ function addMarker(map, coordinates, title, icon, description){
     infoWindow.open(map, marker);
   });	
 }
+
+google.charts.load('current', {'packages':['line']});
+google.charts.setOnLoadCallback(drawTemperatureChart);
+
+/** Draws the Ithaca average monthly temperature chart. */
+function drawTemperatureChart() {
+
+  var data = new google.visualization.DataTable();
+  data.addColumn('string', 'Month');
+  data.addColumn('number', 'High');
+  data.addColumn('number', 'Low');
+
+  data.addRows(tempData);
+
+  const options = {
+    chart: {
+      title: 'Average Monthly Temperatures in Ithaca',
+      subtitle: 'in degrees Celcius',
+    },
+    series: {
+      0: { color: '#DC143C' },
+      1: { color: '#1E90FF' },
+    },
+    backgroundColor: '#FAEBD7',
+    width: 900,
+    height: 500
+  };
+
+  var chart = new google.charts.Line(document.getElementById('weather-chart-container'));
+  chart.draw(data, google.charts.Line.convertOptions(options));
+}
+
+google.charts.load('current', {'packages':['corechart']});
+google.charts.setOnLoadCallback(drawVoteChart);
+
+/** Fetches page vote data and uses it to create a chart. */
+function drawVoteChart() {
+  fetch('/vote-data').then(response => response.json())
+  .then((pageVotes) => {
+    const data = new google.visualization.DataTable();
+    data.addColumn('string', 'Page');
+    data.addColumn('number', 'Votes');
+    Object.keys(pageVotes).forEach((page) => {
+      data.addRow([page, pageVotes[page]]);
+    });
+
+    const options = {
+      title: 'Favorite Page on Website',
+      series: {
+        0: { color: '#DC143C' }
+      },
+      width:800,
+      height:700,
+      backgroundColor: '#FAEBD7',
+    };
+
+    const chart = new google.visualization.ColumnChart(document.getElementById('vote-container'));
+    chart.draw(data, options);
+  });
+}
+
+/** Adds a new vote to the chart showing users' favorite pages of the website. */
+async function addNewVote() {
+  const userVote = document.getElementById('user-vote').value;
+  const params = new URLSearchParams();
+  params.append('userVote', userVote);
+  const add = await fetch('vote-data', {method: 'POST', body: params});
```

Comment:
nit: I think you might want `/vote-data' rather than just `vote-data` here.  I believe that if you ever made this call from a webpage in another folder (like '/sub_dir/page.html' rather than just '/index.html') it might break without the /

Reply:
fixed